### PR TITLE
create virtual node edits

### PR DESCRIPTION
**Pull request by nadinequinr3**
Preview here:

- https://nadine.preview.docs.r3.com/en/platform/corda/5.0/application-networks/creating/mgm/virtual-node.html
- https://nadine.preview.docs.r3.com/en/platform/corda/5.0/application-networks/creating/members/virtual-node.html

### CODE REVIEW

File: content/en/platform/corda/5.0/application-networks/creating/mgm/virtual-node.md [hunk 1]
```diff
@@ -8,18 +8,31 @@ menu:
...
-{{% tab name="PowerShell" %}}
+
+If successful, this request returns the details of the new virtual node as JSON. To save the ID of the virtual node for future use, run the following command, replacing `<holding-identity-ID>` with the ID returned in `holdingIdentity.shortHash` in the received response. For example, `58B6030FABDD`.
+```shell
+export MGM_HOLDING_ID=<holding-identity-ID>
+```
+
+## Create a Virtual Node on Windows
+
+To create a virtual node for the MGM on Windows, run the following commands in PowerShell, changing the X.500 name. The command uses the checksum of the CPI from the repsonse saved when you [uploaded the MGM CPI]({{< relref"./cpi.md#upload-the-cpi" >}}).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/application-networks/creating/mgm/virtual-node.md:32]:
```suggestion
To create a virtual node for the MGM on Windows, run the following commands in PowerShell, changing the X.500 name. The command uses the checksum of the CPI from the response saved when you [uploaded the MGM CPI]({{< relref"./cpi.md#upload-the-cpi" >}}).
```

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

**Review by yarivus — DISMISSED** [review]
LGTM

File: content/en/platform/corda/5.0/application-networks/creating/members/cpi.md
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0/application-networks/creating/members/cpi.md:1]: We need to define `REST_API_USER` and `REST_API_PASSWORD` before line 38.

We could add these to the list of variables on lines 34-37. 

```powershell
$REST_API_USER="username"
$REST_API_PASSWORD="password"
```

We need the same in bash too + `AUTH_INFO` is missing from the bash example.

Line 38 results in an error, replacing it with this fixes it:
```powershell
$AUTH_INFO = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("${REST_API_USER}:${REST_API_PASSWORD}" -f $username,$password)))
```

A similar error happens for line 54. This is the fix:

```powershell
$REST_API_URL="https://${REST_API_HOST}:${REST_API_PORT}/api/v1"
```

What does the `-Force` parameter do on line 70? It looks aggressive, not sure if it is needed.

`WORK_DIR` is set to a different value in bash/powershell examples. We should be consistent. Would be nice if it avoided suggesting a location like home or the desktop. Just make the directory anywhere and point the variable at it.

The preview does not match the content of this PR at this point.

At lines 116/129, we need a keystore. Have we covered generating one by this point in the docs?

Comment (nadinequinr3) [reply content/en/platform/corda/5.0/application-networks/creating/members/cpi.md:1]: Thanks @josephzunigadaly. All of the above should be addressed now except for the query about generating a keystore. Is this something the network operator would do after receiving a key from the developer?

**Comment by colin-hillis** [comment]
@yarivus in fairness, Joseph's edits came in late on Friday evening after regular hours. Nadine's first commit on the associated corrections was 9am this morning, as can be seen from commits above (before your prompt). It doesn't get much faster than that.

**Review by yarivus — APPROVED** [review]
LGTM

File: content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md [hunk 2]
```diff
@@ -9,17 +9,30 @@ menu:
...
-{{% /tab %}}
-{{% tab name="PowerShell" %}}
+
+If successful, this request returns the details of the new virtual node as JSON. To save the ID of the virtual node for future use, run the following command, replacing `<holding-identity-ID>` with the ID returned in `holdingIdentity.shortHash` in the received response. For example, `58B6030FABDD`.
+
+```shell
+export MGM_HOLDING_ID=<holding-identity-ID>
+```
+
+## Create a Virtual Node on Windows
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md:32]: Creating a virtual node has changed to an async operation. The command already in this section will start the process of creating a vnode. The response will contain an ID representing the work of creating that vnode. We need to add an extra request to check the status of the work.

Line 45 no longer works and needs to be removed:
```powershell
$HOLDING_ID = $VIRTUAL_NODE_RESPONSE.holdingIdentity.shortHash  <- remove this
```

It is replaced with a process. This request fetches the status of the work:

```powershell
$VIRTUAL_NODE_RESPONSE_STATUS = Invoke-RestMethod -SkipCertificateCheck  -Headers @{Authorization=("Basic {0}" -f $AUTH_INFO)} -Uri "$REST_API_URL/virtualnode/status/$($VIRTUAL_NODE_RESPONSE.requestId)" -Method Get
```

This request will return a json object with `status = "SUCCEEDED"` once the operation is complete. We may need to call this multiple times until we see `SUCCEEDED`. In our own test code, we loop while the status is `ACCEPTED` or `IN_PROGRESS`. We treat `SUCCEEDED`, `FAILED` or `ABORTED` as end states and stop checking when we see them.

Once we have success, we can read the Holding ID from `resourceId` of that status response. 

```powershell
$HOLDING_ID = $VIRTUAL_NODE_RESPONSE_STATUS.resourceId
```

**At the moment, the request ID and holding ID are the same but we need to be ready for a future where they are different so we should avoid using the request ID as a holding ID in these examples.**

The bash example needs a similar update.

**Review by yarivus — APPROVED** [review]
LGTM

File: content/en/platform/corda/5.0/application-networks/managing/member-suspension/_index.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ The commands shown in this section, use the following variables:
 * `REST_API_PASSWORD` — your password for invoking the REST API.
 * `AUTH_INFO` — the authentication information if using PowerShell. You can set this as follows:
    ```
-   $AUTH_INFO = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("admin:admin" -f $username,$password)))
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0/application-networks/managing/member-suspension/_index.md:27]: Should we use admin/admin as the values for username and password when setting environment variables?

We need to tell the end user what the default values are somewhere. Do we have a page that tells people how to change them to something secure when setting up for the first time?

**Review by yarivus — APPROVED** [review]
LGTM

File: content/en/platform/corda/5.0/application-networks/creating/members/cpi.md
Comment (nadinequinr3) [reply content/en/platform/corda/5.0/application-networks/creating/members/cpi.md:1]: So, we probably need a new set of steps before https://corda5.preview.docs.r3.com/en/platform/corda/5.0/application-networks/creating/members/cpi.html#create-the-cpi-file to describe how to build a keystore?

I would also be of the opinion that we should be aiming for cmplete examples.

File: content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md (hunk 2 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md:38]: @josephzunigadaly I've made a stab at this. Can you review please before I make a similar change to the MGM section. Thanks

File: content/en/platform/corda/5.0/application-networks/managing/member-suspension/_index.md (hunk 3 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0/application-networks/managing/member-suspension/_index.md:27]: No, we probably shouldn't ;) 
I think this is where they set up the credentials: https://corda5.preview.docs.r3.com/en/platform/corda/5.0/deploying-operating/deploying/deploying.html#database

**Review by yarivus — DISMISSED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — DISMISSED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM - clean merge

**Review by yarivus — DISMISSED** [review]
LGTM - clean merge

File: content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md [hunk 4]
```diff
@@ -11,33 +11,53 @@ section_menu: corda5
...
        cpiFileChecksum = $CPI_STATUS_RESPONSE.cpiFileChecksum
        x500Name = $X500_NAME
     }
 })
+```
+Check that the virtual node was created successfully by running the following:
 
-$HOLDING_ID = $VIRTUAL_NODE_RESPONSE.holdingIdentity.shortHash
+```shell
+$VIRTUAL_NODE_RESPONSE_STATUS = Invoke-RestMethod -SkipCertificateCheck  -Headers [@{Authorization=("Basic](https://github.com/{Authorization=("Basic) {0}" -f $AUTH_INFO)} -Uri "$REST_API_URL/virtualnode/status/$($VIRTUAL_NODE_RESPONSE.requestId)" -Method Get
```
Comment (josephzunigadaly) [inline content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md:56]:
```suggestion
$VIRTUAL_NODE_RESPONSE_STATUS = Invoke-RestMethod -SkipCertificateCheck  -Headers @{Authorization=("Basic {0}" -f $AUTH_INFO)} -Uri "$REST_API_URL/virtualnode/status/$($VIRTUAL_NODE_RESPONSE.requestId)" -Method Get
```

**Review by yarivus — DISMISSED** [review]
LGTM

File: content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md [hunk 5]
```diff
@@ -11,33 +11,53 @@ section_menu: corda5
...
+
+```shell
+export MGM_HOLDING_ID = <resource-ID>
 ```
-{{% /tab %}}
-{{% tab name="PowerShell" %}}
+
+## Create a Virtual Node on Windows
+
+To create a virtual node for a member on Windows, run the following commands in PowerShell, changing the X.500 name. The command uses the checksum of the CPI from the repsonse saved when you [uploaded the member CPI]({{< relref"./cpi.md#upload-the-cpi" >}}):
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md:42]:
```suggestion
To create a virtual node for a member on Windows, run the following commands in PowerShell, changing the X.500 name. The command uses the checksum of the CPI from the response saved when you [uploaded the member CPI]({{< relref"./cpi.md#upload-the-cpi" >}}):
```

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

**Review by paulmoloneyr3 — APPROVED** [review]

**Review by yarivus — COMMENTED** [review]
thx @nadinequinr3

**Pull request merged**